periodic/periodic.go

```diff
@@ -117,7 +118,7 @@ func newPeriodicRunner(opts *RunnerOptions) *periodicRunner {
 	if r.Resolution <= 0 {
 		r.Resolution = DefaultRunnerOptions.Resolution
 	}
-	if r.Duration <= 0 {
+	if r.Duration < 0 {
```

Comment:
this breaks the tests which relied on uninitialized duration to get the default duration